### ISSUE TITLE
Prototype Iceberg Versioning

### ISSUE THREAD

**Issue by averyniceday** [knowledgesystems/pipelines-scrum#979]
## Done Condition (What do we need? Why do we need it? Keep this is small as possible!)
Recommendations/Testing written up for review

Fundamental questions to answer:
> How does versioning work with Iceberg?
> How to interact with Iceberg file?
> Where does Dremio fit in?

To think about for implementation
> Pros/Cons of each approach (performance, ease of maintenance, storage cost)
> How will this plug into the MPATH proxy? Does the proxy need to write out a file somewhere or run a command to insert into Iceberg?
> What type of query will MPATH proxy need to run (GoLang Support)?

## Technical Description (How are we going to achieve the above)
JSON stored at **individual** sample level

**Comment by sheridancbio** [comment]
The Iceberg file format includes complex snapshotting and history logging, much like git (or any modern revision control system) formatted files. The Apache Spark datasource engine api is given as an example of one history navigation tool here: https://iceberg.apache.org/docs/latest/spark-queries/#time-travel

**Comment by sheridancbio** [comment]
Dremio hosts a website called _Subsurface community_ .. and this page there introduces iceberg and mentions Dremio support tools which expose iceberg functionality:
https://www.dremio.com/subsurface/introduction-to-apache-iceberg-using-spark/
The tutorial exercises focus on using spark.

A quote:
>Apache Iceberg tables can also be easily used with the Dremio Cloud platform. With Dremio Sonar you can efficiently query Iceberg tables and Dremio Arctic can serve as your Iceberg catalog, enabling git-like semantics (branching/merging) when working with Iceberg tables. You can also explore the structure of Apache Iceberg tables using Dremio.

**Comment by sheridancbio** [comment]
Using directions/examples from here:
https://www.dremio.com/subsurface/introduction-to-apache-iceberg-using-spark/
I was able to create an iceberg file for a single sample ... storing the various top level dictionary key name/key value pairs as JSON values stored in string type fields.

I inserted a large fetched sample (P-0076932-T02-IM7) into the table. This sample is approximately 3.5MB, and a warning was issued:
22/11/03 18:50:32 WARN TaskSetManager: Stage 4 contains a task of very large size (3415 KiB). The maximum recommended task size is 1000 KiB.
Time taken: 1.728 seconds

Hovever, the import seems to have succeeded however
